src/borg/repository.py

```diff
@@ -1475,6 +1476,20 @@ def delete_segment(self, segment):
         except FileNotFoundError:
             pass
 
+    def clear_empty_dirs(self):
+        "Delete empty data dirs, i.e those with no segment files"
+        data_dir = os.path.join(self.path, 'data')
+        segment_dirs = [f.path for f in os.scandir(data_dir) if f.is_dir()]
+        for segment_dir in segment_dirs:
+            if len([f for f in os.scandir(segment_dir)]) == 0:
```

Comment:
simpler: `len(list(os.scandir(segment_dir)))`?

Reply:
Removing it all together, relying on behaviour of `os.rmdir` as suggested